ql/termstructures/globalbootstrap.hpp

```diff
@@ -267,34 +283,44 @@ template <class Curve> void GlobalBootstrap<Curve>::calculate() const {
             ts_->interpolator_.interpolate(ts_->times_.begin(), ts_->times_.end(), ts_->data_.begin());
     }
 
+    // setup guess
+    const Size numberBounds = ts_->times_.size() - 1;
+    Array additionalGuesses;
+    if (additionalVariables_) {
+        additionalGuesses = additionalVariables_->initialize(validCurve_);
+    }
+    Array guess(numberBounds + additionalGuesses.size());
+    for (Size i = 0; i < numberBounds; ++i) {
+        // just pass zero as the first alive helper, it's not used in the standard QL traits anyway
+        // update ts_->data_ since Traits::guess() usually depends on previous values
+        Traits::updateGuess(ts_->data_, Traits::guess(i + 1, ts_, validCurve_, 0), i + 1);
+        guess[i] = Traits::transformInverse(ts_->data_[i + 1], i + 1, ts_);
+    }
+    std::copy(additionalGuesses.begin(), additionalGuesses.end(), guess.begin() + numberBounds);
+
     // setup cost function
     SimpleCostFunction cost([&](const Array& x) {
-        for (Size i = 0; i < x.size(); ++i) {
+        for (Size i = 0; i < numberBounds; ++i) {
             Traits::updateGuess(ts_->data_, Traits::transformDirect(x[i], i + 1, ts_), i + 1);
         }
         ts_->interpolation_.update();
-        Array result(numberHelpers_);
+        if (additionalVariables_) {
+            additionalVariables_->update(Array(x.begin() + numberBounds, x.end()));
+        }
```

Comment:
One can probably figure it out, but it might be helpful to have here some comment on how `x` is part helpers and part additional variables and needs to update them differently.

Reply:
Done